### PR TITLE
bpo-46016: fcntl module add FreeBSD's F_DUP2FD_CLOEXEC flag support.

### PR DESCRIPTION
**Pull request by devnexen**
https://bugs.python.org/issue46016

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2021-12-08-19-15-03.bpo-46016.s9PuyF.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+- Adding F_DUP2FD_CLOEXEC flag from FreeBSD into the fcntl module.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2021-12-08-19-15-03.bpo-46016.s9PuyF.rst]: I would be nice to document the constant somewhere in Doc/library/fcntl.rst:
https://docs.python.org/dev/library/fcntl.html

You can also document the new constant in What's New in Python 3.11, add a new "fcntl" section to:
https://docs.python.org/dev/whatsnew/3.11.html#improved-modules

**Comment by ronaldoussoren** [comment]
Similar functionality on Linux is exposed through ``os.dup`` (the inherited flag, see <https://docs.python.org/3.11/library/os.html?highlight=os.dup2#os.dup2>). It might be better to do the same for FreeBSD (and macOS) as well.

**Comment by tiran** [comment]
The bugs.python.org ticket for this PR is empty. Please update the ticket with an explanation what you are trying to archive.

**Comment by vstinner** [comment]
Merged, thanks. Adding 2 constants is fine.

It would be great to document properly each constant in the fcntl doc (not only these 2 constants), but it can be done in a separated PR.

os.dup2() has a `fcntl(fd, F_DUP2FD_CLOEXEC, fd2)` implementation. Is it used on FreeBSD?

**Comment by vstinner** [comment]
> Similar functionality on Linux is exposed through os.dup (the inherited flag, see https://docs.python.org/3.11/library/os.html?highlight=os.dup2#os.dup2). It might be better to do the same for FreeBSD (and macOS) as well.

F_DUP2FD / F_DUP2FD_CLOEXEC is more about os.dup2(), no? As I wrote, F_DUP2FD_CLOEXEC is already used by os.dup2() if available. I dont know if it is used on macOS. I no longer have access to a macOS machine to develop Python on it.

**Pull request merged**